pkg/rancher/migrations.go

```diff
@@ -28,13 +34,52 @@ const (
 	forceUpgradeLogoutConfig                  = "forceupgradelogout"
 	forceLocalSystemAndDefaultProjectCreation = "forcelocalprojectcreation"
 	forceSystemNamespacesAssignment           = "forcesystemnamespaceassignment"
+	migrateFromMachineToPlanSecret            = "migratefrommachinetoplanesecret"
 	rancherVersionKey                         = "rancherVersion"
 	projectsCreatedKey                        = "projectsCreated"
 	namespacesAssignedKey                     = "namespacesAssigned"
 	caSecretName                              = "tls-ca-additional"
 	caSecretField                             = "ca-additional.pem"
+	capiMigratedKey                           = "capiMigrated"
 )
 
+func runMigrations(wranglerContext *wrangler.Context) error {
+	if err := forceUpgradeLogout(wranglerContext.Core.ConfigMap(), wranglerContext.Mgmt.Token(), "v2.6.0"); err != nil {
+		return err
+	}
+
+	if err := forceSystemAndDefaultProjectCreation(wranglerContext.Core.ConfigMap(), wranglerContext.Mgmt.Cluster()); err != nil {
+		return err
+	}
+
+	if features.MCM.Enabled() {
+		if err := forceSystemNamespaceAssignment(wranglerContext.Core.ConfigMap(), wranglerContext.Mgmt.Project()); err != nil {
+			return err
+		}
+	}
+
+	if features.EmbeddedClusterAPI.Enabled() {
+		if err := addWebhookConfigToCAPICRDs(wranglerContext.CRD.CustomResourceDefinition()); err != nil {
+			return err
+		}
+	}
+
+	if features.RKE2.Enabled() {
```

Comment:
given this traverses all the management clusters do we want to gate this behind a version check?

Reply:
There is a configmap check in `migrateCAPIMachineLabelsAndAnnotationsToPlanSecret` to ensure this migration only happens once.